PureJava/org.eclipse.nebula.visualization.xygraph/src/org/eclipse/nebula/visualization/xygraph/figures/Axis.java

```diff
@@ -286,35 +313,29 @@ protected void paintClientArea(final Graphics graphics) {
 
 	/**
 	 * @return Range that reflects the minimum and maximum value of all traces
-	 *         on this axis. Returns <code>null</code> if there is no trace data
-	 *         or all trace ranges have NaNs or infinities.
+	 *         on this axis. Returns <code>null</code> if there is no trace
+	 *         data.
 	 */
 	public Range getTraceDataRange() {
 		double low = Double.POSITIVE_INFINITY;
 		double high = Double.NEGATIVE_INFINITY;
-		final boolean positiveOnly = isLogScaleEnabled();
```

Comment:
Yes, you could override this as upstream needs fixing properly anyway, it will be best to do it once properly.